numba/core/typing/npydecl.py

```diff
@@ -455,7 +455,7 @@ def parse_shape(shape):
     if isinstance(shape, types.Integer):
         ndim = 1
     elif isinstance(shape, (types.Tuple, types.UniTuple)):
-        if all(isinstance(s, types.Integer) for s in shape):
+        if all(isinstance(s, (types.Integer, types.IntEnumMember)) for s in shape):
```

Comment:
Although this file is not automatically checked by flake8 (it should be, but needs fixing and removing from the ignore list) this line is over-long (more than 80 chars) - perhaps this could be resolved e.g. by:

```suggestion
        int_tys = (types.Integer, types.IntEnumMember)
        if all(isinstance(s, int_tys) for s in shape):
```

Reply:
Sounds good, I'll add that!